CoreScriptsRoot/CoreScripts/ContextActionTouch.lua

```diff
@@ -171,6 +208,65 @@ function createNewButton(actionName, functionInfoTable)
 	return contextButton
 end
 
+function getButtonPosition(index, buttonSize)
+	local complete = false
+	local position, occupyingPosition
+
+	while not complete do
+		if buttonPositionsOccupied[index] then
+			index = index + 1
+		else
+			local info = buttonPositionTable[index]
+
+			-- Find the rotation and distance relative to jump button
+			local rotation, distance
+
+			if info ~= nil then
+				-- Use preset position
+				rotation, distance = unpack(info)
+			else
+				-- Auto generate position based on preset arcs
+				local arc = 0
+				local positionOnArc = 0
+				local totalInArc = 0
+				local indexOnArcs = index - #buttonPositionTable - 1
+				
+				local total = 0
+				for arcIndex=1, #buttonArcs do
+					local numInArc = buttonArcs[arcIndex]
+					local prevTotal = total
+					total = total + numInArc
+
+					if total > indexOnArcs then
+						arc = arcIndex
+						positionOnArc = indexOnArcs - prevTotal
+						totalInArc = numInArc
+						break
+					end
+				end
+
+				rotation = 90 + 90 * ((positionOnArc)/(totalInArc-1))
+				distance = firstArcDistance + arc-1
+			end
+
+			-- Calculate position
+			local buttonBuffer = jumpButtonSize * 1.2
+
+			local pos = jumpButtonPosition + UDim2.new(
+				0,
+				jumpButtonSize/2 + math.cos(math.rad(-rotation)) * buttonBuffer * distance - buttonSize.X.Offset/2,
+				0,
+				jumpButtonSize/2 + math.sin(math.rad(-rotation)) * buttonBuffer * distance - buttonSize.Y.Offset/2
+			)
+
+			position, occupyingPosition = pos, index
+			complete = true
```

Comment:
We can get rid of this 'complete' variable if we move out the search logic mentioned above